pkg/queue/instance.go

```diff
@@ -30,6 +30,7 @@ type Instance interface {
 	Push(task Task)
 	// Run the loop until a signal on the channel
 	Run(<-chan struct{})
+	Size() int
```

Comment:
I would really love to see 'queue' refactored and cleaned up... I thought it's no longer used with the event based interface - do you understand where/why is it still needed ?

Reply:
Its used pretty much everywhere. Any events that come from k8s get put in the queue. That is how k8s implements their controllers aswell